parquet-common/src/main/java/org/apache/parquet/type/Float16.java

```diff
@@ -0,0 +1,339 @@
+   * If either of the values is NaN, the result is false.
+   *
+   * @param x The first half-precision value
+   * @param y The second half-precision value
+   *
+   * @return True if x is less than y, false otherwise
+   */
+  public static boolean less(short x, short y) {
+    if (isNaN(x)) return false;
+    if (isNaN(y)) return false;
+    return ((x & SIGN_MASK) != 0 ? 0x8000 - (x & 0xffff) : x & 0xffff) <
+      ((y & SIGN_MASK) != 0 ? 0x8000 - (y & 0xffff) : y & 0xffff);
+  }
+
+  /**
+   * Returns true if the first half-precision float value is greater (larger
+   * toward positive infinity) than the second half-precision float value.
+   * If either of the values is NaN, the result is false.
+   *
+   * @param x The first half-precision value
+   * @param y The second half-precision value
+   *
+   * @return True if x is greater than y, false otherwise
+   */
+  public static boolean greater(short x, short y) {
+    if (isNaN(x)) return false;
+    if (isNaN(y)) return false;
+    return ((x & SIGN_MASK) != 0 ? 0x8000 - (x & 0xffff) : x & 0xffff) >
+      ((y & SIGN_MASK) != 0 ? 0x8000 - (y & 0xffff) : y & 0xffff);
+  }
+
+  /**
+   * <p>Compares the two specified half-precision float values. The following
+   * conditions apply during the comparison:</p>
+   *
+   * <ul>
+   * <li>{@link #NaN} is considered by this method to be equal to itself and greater
+   * than all other half-precision float values (including {@code #POSITIVE_INFINITY})</li>
+   * <li>{@link #POSITIVE_ZERO} is considered by this method to be greater than
+   * {@link #NEGATIVE_ZERO}.</li>
+   * </ul>
+   *
+   * @param x The first half-precision float value to compare.
+   * @param y The second half-precision float value to compare
+   *
+   * @return  The value {@code 0} if {@code x} is numerically equal to {@code y}, a
+   *          value less than {@code 0} if {@code x} is numerically less than {@code y},
+   *          and a value greater than {@code 0} if {@code x} is numerically greater
+   *          than {@code y}
+   *
+   */
+  public static int compare(short x, short y) {
+    if (less(x, y)) return -1;
+    if (greater(x, y)) return 1;
+    // Collapse NaNs, akin to halfToIntBits(), but we want to keep
+    // (signed) short value types to preserve the ordering of -0.0
+    // and +0.0
+    short xBits = isNaN(x) ? NaN : x;
+    short yBits = isNaN(y) ? NaN : y;
+    return (xBits == yBits ? 0 : (xBits < yBits ? -1 : 1));
+  }
+
+  /**
+   * Converts byte[] representation in Little-Endian
+   * of a half-precision float value to its half-precision float value.
+   * Throw InvalidFloat16ValueException if length of bytes is not equal to 2.
+   *
+   * @param bytes byte[] representation of a half-precision float value in Little-Endian
+   * @return A half-precision float value
+   */
+  public static short fromBytesLittleEndian(byte[] bytes) {
+    if (bytes.length != 2) {
+      throw new InvalidFloat16ValueException(String.valueOf(bytes));
```

Comment:
Does `String.valueOf` reflect the length of the input bytes? If not, we probably need to add it to the error message.